File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3SiriDownstreamSubscriptionTopic.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3SiriDownstreamSubscriptionTopic {
    /// <summary>
    /// Gets or Sets LineRef
    /// </summary>
    [DataMember(Name="line_ref", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "line_ref")]
    public string LineRef { get; set; }

    /// <summary>
    /// Gets or Sets DirectionRef
    /// </summary>
    [DataMember(Name="direction_ref", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "direction_ref")]
    public int? DirectionRef { get; set; }

    /// <summary>
    /// Gets or Sets RouteType
    /// </summary>
    [DataMember(Name="route_type", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "route_type")]
    public int? RouteType { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3SiriDownstreamSubscriptionTopic {\n");
      sb.Append("  LineRef: ").Append(LineRef).Append("\n");
      sb.Append("  DirectionRef: ").Append(DirectionRef).Append("\n");
      sb.Append("  RouteType: ").Append(RouteType).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: dart/lib/model/v3_siri_downstream_subscription_topic.dart
part of swagger.api;

class V3SiriDownstreamSubscriptionTopic {
  
  String lineRef = null;

  int directionRef = null;
  //enum directionRefEnum {  1,  2,  5,  10,  16,  32,  65,  130,  };

  int routeType = null;
  //enum routeTypeEnum {  0,  1,  2,  3,  4,  };

  V3SiriDownstreamSubscriptionTopic();

  @override
  String toString() {
    return 'V3SiriDownstreamSubscriptionTopic[lineRef=$lineRef, directionRef=$directionRef, routeType=$routeType, ]';
  }

  V3SiriDownstreamSubscriptionTopic.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    lineRef = json['line_ref'];
    directionRef = json['direction_ref'];
    routeType = json['route_type'];
  }

  Map<String, dynamic> toJson() {
    return {
      'line_ref': lineRef,
      'direction_ref': directionRef,
      'route_type': routeType
     };
  }

  static List<V3SiriDownstreamSubscriptionTopic> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3SiriDownstreamSubscriptionTopic>() : json.map((value) => new V3SiriDownstreamSubscriptionTopic.fromJson(value)).toList();
  }

  static Map<String, V3SiriDownstreamSubscriptionTopic> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3SiriDownstreamSubscriptionTopic>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3SiriDownstreamSubscriptionTopic.fromJson(value));
    }
    return map;
  }
}
